gitlab/resource_gitlab_project_badge_test.go

```diff
@@ -0,0 +1,178 @@
+package gitlab
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"github.com/xanzy/go-gitlab"
+)
+
+func TestAccGitlabProjectBadge_basic(t *testing.T) {
+	var badge gitlab.ProjectBadge
+	rInt := acctest.RandInt()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGitlabProjectBadgeDestroy,
+		Steps: []resource.TestStep{
+			// Create a project and badge
+			{
+				Config: testAccGitlabProjectBadgeConfig(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabProjectBadgeExists("gitlab_project_badge.foo", &badge),
+					testAccCheckGitlabProjectBadgeAttributes(&badge, &testAccGitlabProjectBadgeExpectedAttributes{
+						LinkURL:  fmt.Sprintf("https://example.com/badge-%d", rInt),
+						ImageURL: fmt.Sprintf("https://example.com/badge-%d.svg", rInt),
+					}),
+				),
+			},
+			// Update the project badge
+			{
+				Config: testAccGitlabProjectBadgeUpdateConfig(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabProjectBadgeExists("gitlab_project_badge.foo", &badge),
+					testAccCheckGitlabProjectBadgeAttributes(&badge, &testAccGitlabProjectBadgeExpectedAttributes{
+						LinkURL:  fmt.Sprintf("https://example.com/badge-%d", rInt),
+						ImageURL: fmt.Sprintf("https://example.com/badge-%d.svg", rInt),
+					}),
+				),
+			},
+		},
+	})
+}
+
+func TestAccGitlabProjectBadge_import(t *testing.T) {
+	rInt := acctest.RandInt()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGitlabProjectBadgeDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGitlabProjectBadgeConfig(rInt),
+			},
```

Comment:
These tests can be combined. See the first bullet here: https://www.terraform.io/docs/extend/resources/import.html#recommendations-for-import

```suggestion
```

Reply:
@armsnyder I added the docs and combined tests in one single function. If you find something else, please let me know